src/utils/event-manager.ts

```diff
@@ -22,29 +66,101 @@ export class EventManager extends Construct {
    * The event bus to register new rules with.
    * @attribute
    */
-  private readonly eventBus: IEventBus;
+  public readonly eventBus: IEventBus;
+
+  public readonly applicationPlaneEventSource: string = 'testApplicationPlaneEventSource';
+  public readonly controlPlaneEventSource: string = 'testControlPlaneEventSource';
+  // sensible defaults so they are not required when instantiating control plane
+  readonly eventMetadata: EventMetadata = {
+    onboardingRequest: {
+      source: [this.controlPlaneEventSource],
+      detailType: ['onboardingRequest'],
+    },
+    onboardingSuccess: {
+      source: [this.applicationPlaneEventSource],
+      detailType: ['onboardingSuccess'],
+    },
+    onboardingFailure: {
+      source: [this.applicationPlaneEventSource],
+      detailType: ['onboardingFailure'],
+    },
+    offboardingRequest: {
+      source: [this.controlPlaneEventSource],
+      detailType: ['offboardingRequest'],
+    },
+    offboardingSuccess: {
+      source: [this.applicationPlaneEventSource],
+      detailType: ['offboardingSuccess'],
+    },
+    offboardingFailure: {
+      source: [this.applicationPlaneEventSource],
+      detailType: ['offboardingFailure'],
+    },
+    provisionSuccess: {
+      source: [this.applicationPlaneEventSource],
+      detailType: ['provisionSuccess'],
+    },
+    provisionFailure: {
+      source: [this.applicationPlaneEventSource],
+      detailType: ['provisionFailure'],
+    },
+    deprovisionSuccess: {
+      source: [this.applicationPlaneEventSource],
+      detailType: ['deprovisionSuccess'],
+    },
+    deprovisionFailure: {
+      source: [this.applicationPlaneEventSource],
+      detailType: ['deprovisionFailure'],
+    },
+    billingSuccess: {
+      source: [this.controlPlaneEventSource],
+      detailType: ['billingSuccess'],
+    },
+    billingFailure: {
+      source: [this.controlPlaneEventSource],
+      detailType: ['billingFailure'],
+    },
+    activateRequest: {
+      source: [this.controlPlaneEventSource],
+      detailType: ['activateRequest'],
+    },
+    activateSuccess: {
+      source: [this.applicationPlaneEventSource],
+      detailType: ['activateSuccess'],
+    },
+    activateFailure: {
+      source: [this.applicationPlaneEventSource],
+      detailType: ['activateFailure'],
+    },
+    deactivateRequest: {
+      source: [this.controlPlaneEventSource],
+      detailType: ['deactivateRequest'],
+    },
+    deactivateSuccess: {
+      source: [this.applicationPlaneEventSource],
+      detailType: ['deactivateSuccess'],
+    },
+    deactivateFailure: {
+      source: [this.applicationPlaneEventSource],
+      detailType: ['deactivateFailure'],
+    },
+  };
+
+  private readonly rules: { [key: string]: Rule } = {};
+
   constructor(scope: Construct, id: string, props: EventManagerProps) {
     super(scope, id);
     this.eventBus = props.eventBus;
+
+    for (const [key, value] of Object.entries(this.eventMetadata)) {
```

Comment:
I would prefer to create rules for only those events that we've subscribed to. I might be missing something here, so let's discuss. Again, suggestion in TS Playground.

Reply:
updated to use getOrCreate function